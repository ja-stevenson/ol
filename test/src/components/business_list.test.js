import React from 'react';
import { shallow, mount, render } from 'enzyme';
import renderer from 'react-test-renderer';

import BusinessList from '../../../src/components/business_list'

describe('BusinessList', function() {
  it('should render without throwing an error', function() {
    expect(shallow(<BusinessList
          activePage={1}
          updateCurrentPage={() => {}}
          pages={1000}
          businessList={[]}
          onBusinessSelect={() => {}}
        />).contains(<br />)).toBe(true);
  });

  it('should be selectable by class "bus-list"', function() {
    expect(shallow(<BusinessList
          activePage={1}
          updateCurrentPage={() => {}}
          pages={1000}
          businessList={[]}
          onBusinessSelect={() => {}}
        />).is('.bus-list')).toBe(true);
  });

  it('should mount in a full DOM', function() {
    expect(mount(<BusinessList
          activePage={1}
          updateCurrentPage={() => {}}
          pages={1000}
          businessList={[]}
          onBusinessSelect={() => {}}
        />).find('.bus-list').length).toBe(1);
  });

  it('should render to static HTML', function() {
    expect(render(<BusinessList
          activePage={1}
          updateCurrentPage={() => {}}
          pages={1000}
          businessList={[]}
          onBusinessSelect={() => {}}
        />).text()).toEqual('firstprev123456789nextlast');
  });

  it('renders details', () => {
    const component = renderer.create(<BusinessList
          activePage={1}
          updateCurrentPage={() => {}}
          pages={1000}
          businessList={[]}
          onBusinessSelect={() => {}}
        />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});