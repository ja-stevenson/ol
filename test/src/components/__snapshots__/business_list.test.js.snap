exports[`BusinessList renders details 1`] = `
<div
  className="bus-list">
  <ul
    className="list-group" />
  <br />
  <ul
    className="pagination">
    <li
      className="disabled"
      onClick={[Function]}>
      <a
        href="#">
        first
      </a>
    </li>
    <li
      className="disabled"
      onClick={[Function]}>
      <a
        href="#">
        prev
      </a>
    </li>
    <li
      className="active"
      onClick={[Function]}>
      <a
        href="#">
        1
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        2
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        3
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        4
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        5
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        6
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        7
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        8
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        9
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        next
      </a>
    </li>
    <li
      className=""
      onClick={[Function]}>
      <a
        href="#">
        last
      </a>
    </li>
  </ul>
  <a
    className="ol-img"
    href="http://www.ownlocal.com/">
    <img
      src="https://encrypted-tbn3.gstatic.com/images?q=tbn:ANd9GcTJAtzGDIuAddcpYED5NIhIcKu10erWD74zYdoGHE-lWGDOvvf-" />
  </a>
</div>
`;
